src/worker/ingestion/process-event.ts

```diff
@@ -389,63 +412,68 @@ export class EventsProcessor {
             firstSeen = otherPerson.created_at
         }
 
-        await this.db.updatePerson(mergeInto, { created_at: firstSeen, properties: mergeInto.properties })
+        let kafkaMessages: ProducerRecord[] = []
 
-        // Merge the distinct IDs
-        await this.db.postgresQuery(
-            'UPDATE posthog_cohortpeople SET person_id = $1 WHERE person_id = $2',
-            [mergeInto.id, otherPerson.id],
-            'updateCohortPeople'
-        )
         let failedAttempts = totalMergeAttempts
-        let shouldRetryAliasOperation = false
 
         // Retrying merging up to `MAX_FAILED_PERSON_MERGE_ATTEMPTS` times, in case race conditions occur.
         // An example is a distinct ID being aliased in another plugin server instance,
         // between `moveDistinctId` and `deletePerson` being called here
         // – in such a case a distinct ID may be assigned to the person in the database
         // AFTER `otherPersonDistinctIds` was fetched, so this function is not aware of it and doesn't merge it.
-        // That then causeds `deletePerson` to fail, because of foreign key constraints –
+        // That then causes `deletePerson` to fail, because of foreign key constraints –
         // the dangling distinct ID added elsewhere prevents the person from being deleted!
         // This is low-probability so likely won't occur on second retry of this block.
         // In the rare case of the person changing VERY often however, it may happen even a few times,
         // in which case we'll bail and rethrow the error.
-        while (true) {
+        await this.db.postgresTransaction(async (client) => {
```

Comment:
Is it possible we can put this around the whole event handling code instead? i.e. we really want handleIdentify and handleAlias to be atomic

Reply:
It's reasonable to consider expanding how much the transaction wraps in a further PR. 

However, I'd push back against wrapping the entire event handling code on it. We can discuss this further elsewhere as it is a bit outside the scope here but for one @fuziontech was already raising concerns about performance and Postgres becoming a bottleneck if we use transactions too liberally. 

Let's have this discussion elsewhere though as I am indeed planning on pushing more refactors, and we should:

a) ship this PR fast
b) it already includes quite a few changes

Reply:
Yup, for sure it's an improvement already, happy to get in without.

@fuziontech do we have any stats on the number of open transactions we have at the moment? I suspect even wrapping all event handlers in a transactions they will still be very sort lived. If we want to avoid transactions then maybe we can refactor to not require transactions but still maintain integrity?